### PR TITLE
WPS431: allow define custom nested classes whitelist

### LINKED ISSUE #808: WPS431 Found nested class: allow define additional allowed classes
Hi! We are using `graphene-python` package. This library introduces a definition of additional nested classes (for example, `Arguments`, `Input `,..). For now, rule WPS431 prevent these nested classes. I think, that it will be cool to allow to modify `NESTED_CLASSES_WHITELIST` for adding such classes (and any other custom nested classes). Thank you for the linter!

### PR DESCRIPTION
**Pull request by lapshin1988**
## Checklist

- [*] I have double-checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [*] I have created at least one test case for the changes I have made
- [*] I have updated the documentation for the changes I have made
- [*] I have added my changes to the `CHANGELOG.md`

## Related issues
- Closes #808 

Please check added question in "TODO"

### CODE REVIEW

File: tests/test_visitors/test_ast/test_complexity/test_nested/test_nested_classes.py [hunk 1]
```diff
@@ -147,6 +147,34 @@ def test_whitelist_nested_classes(
     assert_errors(visitor, [])
 
 
+@pytest.mark.parametrize('whitelist_name', [
+    *NESTED_CLASSES_WHITELIST,
+    'NestedClass',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_complexity/test_nested/test_nested_classes.py]: Why is `NestedClass` allowed?

File: tests/test_visitors/test_ast/test_complexity/test_nested/test_nested_classes.py [hunk 2]
```diff
@@ -147,6 +147,34 @@ def test_whitelist_nested_classes(
...
+    whitelist_name,
+    code,
+    options,
+    mode,
+):
+    """Testing that it is possible to nest custom whitelisted classes."""
+    tree = parse_ast_tree(mode(code.format(whitelist_name)))
+
+    option_values = options(
+        nested_classes_whitelist=[*NESTED_CLASSES_WHITELIST, 'NestedClass'],
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_complexity/test_nested/test_nested_classes.py]: This should be a default. So, no need to pass it as options.

File: wemake_python_styleguide/options/validation.py [hunk 3]
```diff
@@ -59,6 +59,7 @@ class _ValidatedOptions(object):
     max_asserts: int = attr.ib(validator=[_min_max(min=1)])
     max_access_level: int = attr.ib(validator=[_min_max(min=1)])
     max_attributes: int = attr.ib(validator=[_min_max(min=1)])
+    nested_classes_whitelist: List[str]
```
Comment (sobolevn) [inline wemake_python_styleguide/options/validation.py]: It is a `FrozenSet[str]`, not a list.

File: wemake_python_styleguide/types.py [hunk 4]
```diff
@@ -123,3 +123,4 @@ class or structure.
     max_asserts: int
     max_access_level: int
     max_attributes: int
+    nested_classes_whitelist: List[str]
```
Comment (sobolevn) [inline wemake_python_styleguide/types.py]: The same here.

File: wemake_python_styleguide/options/validation.py (hunk 3 above)
Comment (lapshin1988) [reply wemake_python_styleguide/options/validation.py]: It was first in the first implementation. But flake8 provide a list of strings. I thought that using `List[str]` type annotation will be closer to reality. But ok: I can change annotation to `FrozenSet[str]`. Do it?

**Comment by lapshin1988** [comment]
@sobolevn Sorry, did you see my comments?

**Comment by sobolevn** [comment]
Thanks a lot! I am merging it, I will play around with types to see what fits bests.

P.S. Are you ready for the upcoming https://hacktoberfest.digitalocean.com? We have a lot of stuff planned for this event! Join the celebration 🙂

**Comment by lapshin1988** [comment]
Thank you! About https://hacktoberfest.digitalocean.com - very interesting! Let's do it!😀

**Comment by DjaPy** [comment]
Sorry. Maybe I'm stupid. But I can not get rid of the error. When I crawl a `pydantic` model and configure the class Config nested model, an error occurs

**Comment by DjaPy** [comment]
Sorry. Maybe I'm stupid. But I can not get rid of the error. When I crawl a `pydantic` model and configure the class Config nested model, an error occurs

**Comment by DjaPy** [comment]
Sorry. Maybe I'm stupid. But I can not get rid of the error. When I crawl a `pydantic` model and configure the class Config nested model, an error occurs.

_setup.cfg_
```
nested-classes-whitelist = Meta, Params, Config,
```

_***.py_
```python

class Application(BaseModel):
    id: UUID
    ...

    class Config(object):
        use_enum_values = True
```
_error_
```
***.py:52:5: WPS431 Found nested class: Config
    class Config(object):
```

**Comment by sobolevn** [comment]
What version of `wps` are you using?

**Comment by DjaPy** [comment]
Yes! I don't use last version. Thanks.

**Pull request merged**